nltk/classify/textcat.py

```diff
@@ -144,37 +152,43 @@ def guess_language(self, text):
         self.last_distances = self.lang_dists(text)
         
         return min(self.last_distances, key=self.last_distances.get)
+        #################################################')
+
+    def demo(self):
+        from nltk.corpus import udhr
+
+        langs = ['Kurdish-UTF8', 'Abkhaz-UTF8', 'Farsi_Persian-UTF8',
+                 'Hindi-UTF8', 'Hawaiian-UTF8', 'Russian-UTF8', 'Vietnamese-UTF8',
+                 'Serbian_Srpski-UTF8','Esperanto-UTF8']
+
+        friendly = {'kmr':'Northern Kurdish',
+                    'abk':'Abkhazian',
+                    'pes':'Iranian Persian',
+                    'hin':'Hindi',
+                    'haw':'Hawaiian',
+                    'rus':'Russian',
+                    'vie':'Vietnamese',
+                    'srp':'Serbian',
+                    'epo':'Esperanto'}
         
-def demo():
-    ''' Demo of language guessing using a bunch of UTF-8 encoded
-        text files with snippets of text copied from news websites
-        around the web in different languages '''
-
-    from os import listdir
-    from os.path import isfile
-
-    path = '.'
-    lang_samples = []
-
-    tc = TextCat()
-        
-    for f in listdir(path):
-        if isfile(f):
-            m = regex.match('sample_\w+\.txt', f)
-            if m: lang_samples.append(f)
-                
-    print(lang_samples)
-    for f in lang_samples:
-        cur_sample = open(f, 'rU')
-        cur_data = cur_sample.read()
-        print('Language sample file: ' + f)
-        print('Contents snippet:  ' + cur_data.decode('utf8')[0:140])
-        print('#################################################')
-        print('Language detection: ' + tc.guess_language(cur_data))
-        print('#################################################')
-
-if __name__ == '__main__':
-    import doctest
-    doctest.testmod(optionflags=doctest.NORMALIZE_WHITESPACE |
-    doctest.ELLIPSIS)
-
+        for cur_lang in langs:
+            # Get raw data from UDHR corpus
+            raw_sentences = udhr.sents(cur_lang)
+            rows = len(raw_sentences) - 1
+            cols = map(len, raw_sentences)
+
+            sample = ''
+          
+            # Generate a sample text of the language
+            for i in range(0, rows):
+                cur_sent = ''
+                for j in range(0, cols[i]):
+                    cur_sent += ' ' + raw_sentences[i][j]
+            
+                sample += cur_sent
+          
+            # Try to detect what it is
+            print('Language snippet: ' + sample[0:140] + '...')
+            guess = self.guess_language(sample.encode('utf8'))
```

Comment:
Why is language detection working on bytestrings instead of unicode? It could make sense to work on bytes if there is also some encoding detection, or an ability to handle arbitrary bytes data, but this is not the case here - text is decoded from utf8 in `remove_punctuation` function.